x-pack/plugins/fleet/server/services/artifacts/utils.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ResponseError } from '@elastic/elasticsearch/lib/errors';
+
+interface ErrorWithApiResponseMeta extends Error {
+  meta: ResponseError['meta'];
+}
+
+/**
+ * type assertion.
+ * Checks to see if an error looks to have a `meta` property that holds an `ApiResponse`
+ * @param error
+ */
+export const isErrorWithMeta = (
```

Comment:
we have a `isESClientError` method in `server/errors` that you can use for that :)

Reply:
I can switch over to that one - although it only checks that it an instance of `ResponseError`. I was looking around the types of error elasticsearch client could return (under `node_modules/@elastic/elasticsearch/lib/errors.d.ts`) and some of them don't have `meta` on it, thus why I created this one.

Reply:
@nchaulet I can remove this and use the one you pointed me to if you think its a non-issue to use the `ResponseError` assertion